metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1262,6 +1260,55 @@ class MetalsLanguageServer(
             val msg = s"Invalid arguments: $args. Expecting: $argExample"
             Future.failed(new IllegalArgumentException(msg)).asJavaObject
         }
+      case ServerCommands.NewScalaWorksheet() =>
+        val args = params.getArguments.asScala
+        (args match {
+          case Seq(JsonString.Option(directory), JsonString(name)) =>
```

Comment:
Can the parameters just contain a single object that we would translate to a class using `JsonParser`? We wouldn't need to do the crazy unapplies.

And we could use the kind parameter for everything. There wouldn't need to be separate logic for creating worksheets and classes.

Reply:
@tgodzik 
Maybe it's worth to create full `metals/newScalaFile` endpoint method? I'm not really sure which one is more appropriate in what situations.

Reply:
I think a command is better since it can also work for other clients. Although not 100 % sure

Reply:
I'd be in favor of a `newScalaFile` endpoint since looking at it, there is a lot that is shared. They can also both share the same params and just have kind be either trait, object, class, or worksheet.

Reply:
Made one new command `new-scala-file` instead of previous two, and `MetalsNewScalaFileParams` case class with all it's needed arguments.